src/main/java/com/gmail/goosius/siegewar/playeractions/PlaceBlock.java

```diff
@@ -225,8 +225,10 @@ public static void evaluatePlaceColouredBannerNearTown(Player player,
 														   Town nearbyTown,
 														   Block bannerBlock) throws TownyException {
 
-		//Check whether nearby town has a current or recent siege
-		if (SiegeController.hasSiege(nearbyTown)) {
+		//Check whether nearby town has a current or recent siege or is unsiegeable altogether
+		if (TownMetaDataController.getSiegeImmunityEndTime(nearbyTown) == -1l) {
+			throw new TownyException(Translation.of("msg_err_cannot_start_siege_at_unsiegeable_town"));
```

Comment:
:construction: 
- This is not the right place for the "check if attack is possible" test
- The right place for that logic  is in PlaceBlock.java , at line 271
- And when the logic is placed there, no new lang string will be needed